pykokkos/interface/views.py

```diff
@@ -532,6 +532,10 @@ def asarray(obj, /, *, dtype=None, device=None, copy=None):
     # TODO: proper implementation/design
     # for now, let's cheat and use NumPy asarray() followed
     # by pykokkos from_numpy()
+    if obj in {pk.e, pk.pi, pk.inf, pk.nan}:
+        view = pk.View([1], dtype=pk.float32)
```

Comment:
Regarding the data type issue you mentioned, couldn't you just use the `dtype` argument if it's set to `float64`, and then have it default to `float32` otherwise?